source/_patterns/01-atoms/list-group/demo/list-groups.twig

```diff
@@ -0,0 +1,186 @@
+{# This function loops through the scssColors global variable to make an array of all the available colors. #}
+{% set all_colors = [] %}
+{% for name, rgb in scssColors %}
+  {% set item_classes = {
+    item_color: name
+  } %}
+  {% set all_colors = all_colors|merge([item_classes]) %}
+{% endfor %}
+
+<div class="container">
+  {% embed '@atoms/grid/_grid-1-up.twig' %}
+    {% block column_1 %}
+      <div class="mb-3">
+        <h1 class="text-center">List Group</h1>
+        {% include '@molecules/card/_card-helper.twig' with {
+          card_text: list_group_helper,
+        } %}
+      </div>
+
+      {# Basic examples #}
+      <div class="mb-3">
+        {% include '@molecules/card/_card-helper.twig' with {
+          card_title: 'Basic Examples',
+          card_text: list_group_helper_basic,
+        } %}
+      </div>
+      <div class="row mb-3">
+        <div class="col-sm-6">
+          {% include '@atoms/list-group/_list-group.twig' with list_group_text %}
+        </div>
+        <div class="col-sm-6">
+          {# This block of code is going to appear multiple times. It exists because the _list-group.twig template ultimately needs an object to iterate through to generate the individual items. These code blocks exist to pull data from multiple disperate places (the all_colors array, list-groups.yml, etc.) and combine them into a single object that the template can then process. #}
+          {% set list_color = [] %}
+          {% for item in list_group_text.items %}
+            {% set temp_item = [] | merge(list_group_text.items[loop.index0]) | merge(all_colors[loop.index0]) %}
+            {% set list_color = list_color|merge([temp_item]) %}
```

Comment:
This seems unnecessary, seems like it can be integrated into the line above.

Reply:
Could you clarify how to do that? . The only way I can think to do that would be something like 
`{% set list_color[] = [] | merge(list_group_text.items[loop.index0]) | merge(all_colors[loop.index0]) %}` so that its adding to the array and not overriding it in each iteration of the loop. But this syntax doesn't work.

Reply:
@rloos289, let's do this: Can you distill down the data structures via https://twigfiddle.com/ so I can go hands-on with it?

Reply:
Maybe refactor this whole thing using macros instead?